### PR TITLE
[TEST]  Run pre 6.4 nodes in non-FIPS JVMs

### LINKED ISSUE #32737: :x-pack:qa:full-cluster-restart fails with SecurityException: Keystore has been corrupted or tampered with on FIPS JVM
Relevant log 
```
18:02:02 > Task :x-pack:qa:full-cluster-restart:with-system-key:v6.3.3-SNAPSHOT#oldClusterTestCluster#node0.addToKeystore#xpack.watcher.encryption_key
18:02:02 Task ':x-pack:qa:full-cluster-restart:with-system-key:v6.3.3-SNAPSHOT#oldClusterTestCluster#node0.addToKeystore#xpack.watcher.encryption_key' is not up-to-date because:
18:02:02   Task has not declared any outputs despite executing actions.
18:02:02 	at sun.security.provider.JavaKeyStore.engineLoad(JavaKeyStore.java:658)
18:02:02 	at sun.security.provider.JavaKeyStore$JKS.engineLoad(JavaKeyStore.java:56)
18:02:02 	at sun.security.provider.KeyStoreDelegator.engineLoad(KeyStoreDelegator.java:224)
18:02:02 	at sun.security.provider.JavaKeyStore$DualFormatJKS.engineLoad(JavaKeyStore.java:70)
18:02:02 	at java.security.KeyStore.load(KeyStore.java:1445)
18:02:02 Starting process 'command '/var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8fips/nodes/virtual&&linux/x-pack/qa/full-cluster-restart/with-system-key/build/cluster/v6.3.3-SNAPSHOT#oldClusterTestCluster node0/elasticsearch-6.3.3-SNAPSHOT/bin/elasticsearch-keystore''. Working directory: /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8fips/nodes/virtual&&linux/x-pack/qa/full-cluster-restart/with-system-key/build/cluster/v6.3.3-SNAPSHOT#oldClusterTestCluster node0/cwd Command: /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8fips/nodes/virtual&&linux/x-pack/qa/full-cluster-restart/with-system-key/build/cluster/v6.3.3-SNAPSHOT#oldClusterTestCluster node0/elasticsearch-6.3.3-SNAPSHOT/bin/elasticsearch-keystore add-file xpack.watcher.encryption_key /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8fips/nodes/virtual&&linux/x-pack/qa/full-cluster-restart/src/test/resources/system_key
18:02:02 Successfully started process 'command '/var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8fips/nodes/virtual&&linux/x-pack/qa/full-cluster-restart/with-system-key/build/cluster/v6.3.3-SNAPSHOT#oldClusterTestCluster node0/elasticsearch-6.3.3-SNAPSHOT/bin/elasticsearch-keystore''
18:02:02 warning: ignoring JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8
18:02:02 	at sun.security.util.AnchorCertificates$1.run(AnchorCertificates.java:61)
18:02:02 	at sun.security.util.AnchorCertificates$1.run(AnchorCertificates.java:52)
18:02:02 	at java.security.AccessController.doPrivileged(Native Method)
18:02:02 	at sun.security.util.AnchorCertificates.<clinit>(AnchorCertificates.java:52)
18:02:02 	at sun.security.provider.certpath.AlgorithmChecker.checkFingerprint(AlgorithmChecker.java:214)
18:02:02 	at sun.security.provider.certpath.AlgorithmChecker.<init>(AlgorithmChecker.java:164)
18:02:02 	at sun.security.provider.certpath.AlgorithmChecker.<init>(AlgorithmChecker.java:118)
18:02:02 	at sun.security.validator.SimpleValidator.engineValidate(SimpleValidator.java:157)
18:02:02 	at sun.security.validator.Validator.validate(Validator.java:262)
18:02:02 	at sun.security.validator.Validator.validate(Validator.java:238)
18:02:02 	at sun.security.validator.Validator.validate(Validator.java:207)
18:02:02 	at javax.crypto.JarVerifier.isTrusted(JarVerifier.java:610)
18:02:02 	at javax.crypto.JarVerifier.verifySingleJar(JarVerifier.java:530)
18:02:02 	at javax.crypto.JarVerifier.verifyJars(JarVerifier.java:363)
18:02:02 	at javax.crypto.JarVerifier.verify(JarVerifier.java:289)
18:02:02 	at javax.crypto.JceSecurity.verifyProviderJar(JceSecurity.java:164)
18:02:02 	at javax.crypto.JceSecurity.getVerificationResult(JceSecurity.java:190)
18:02:02 	at javax.crypto.JceSecurity.canUseProvider(JceSecurity.java:204)
18:02:02 	at javax.crypto.SecretKeyFactory.nextSpi(SecretKeyFactory.java:295)
18:02:02 	at javax.crypto.SecretKeyFactory.<init>(SecretKeyFactory.java:121)
18:02:02 	at javax.crypto.SecretKeyFactory.getInstance(SecretKeyFactory.java:160)
18:02:02 	at org.elasticsearch.common.settings.KeyStoreWrapper.createCipher(KeyStoreWrapper.java:288)
18:02:02 	at org.elasticsearch.common.settings.KeyStoreWrapper.decrypt(KeyStoreWrapper.java:336)
18:02:02 	at org.elasticsearch.common.settings.AddFileKeyStoreCommand.execute(AddFileKeyStoreCommand.java:73)
18:02:02 	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
18:02:02 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
18:02:02 	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:79)
18:02:02 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
18:02:02 	at org.elasticsearch.cli.Command.main(Command.java:90)
18:02:02 	at org.elasticsearch.common.settings.KeyStoreCli.main(KeyStoreCli.java:41)
18:02:02 Exception in thread "main" java.lang.SecurityException: Keystore has been corrupted or tampered with
18:02:02 	at org.elasticsearch.common.settings.KeyStoreWrapper.decrypt(KeyStoreWrapper.java:349)
18:02:02 	at org.elasticsearch.common.settings.AddFileKeyStoreCommand.execute(AddFileKeyStoreCommand.java:73)
18:02:02 	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
18:02:02 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
18:02:02 	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:79)
18:02:02 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
18:02:02 	at org.elasticsearch.cli.Command.main(Command.java:90)
18:02:02 	at org.elasticsearch.common.settings.KeyStoreCli.main(KeyStoreCli.java:41)
```

The  test is calling  `elasticsearch-keystore add xpack.watcher.encryption_key -x`

### LINKED ISSUE #32868: java8fips CI build never passes due to 5.6.11-SNAPSHOT bwc task dependency
Job URLs: from http://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java8fips,nodes=virtual&&linux/30
through to http://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java8fips,nodes=virtual&&linux/38 (inclusive)

This build has failed the last 9 builds with:
```
07:53:27 Could not determine the dependencies of task ':x-pack:qa:rolling-upgrade:with-system-key:v5.6.11-SNAPSHOT#upgradedClusterTestRunner'.
07:53:27 > Task with path 'v5.6.11-SNAPSHOT#twoThirdsUpgradedTestCluster#stop' not found in project ':x-pack:qa:rolling-upgrade:with-system-key'.
```

I wonder if its failing because testing FIPS against 5.6.x is not something that makes sense? In which case maybe we need to mute the bwc build against this branch?

### PR DESCRIPTION
**Pull request by jkakavas**
Elasticsearch versions earlier than 6.4.0 cannot properly run in a
FIPS 140 JVM. This commit ensures that we use a non-FIPS JVM for
nodes that we spin up in BWC tests even when we're testing FIPS.

Resolves #32737 

It also reverts e497173 and e64bb48 and as such resolves #32868

### CODE REVIEW

**Review by alpar-t — APPROVED** [review]

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy [hunk 1]
```diff
@@ -177,6 +177,9 @@ class NodeInfo {
             javaVersion = 8
         } else if (nodeVersion.onOrAfter("6.2.0") && nodeVersion.before("6.3.0")) {
             javaVersion = 9
+        } else if (project.inFipsJvm && nodeVersion.onOrAfter("6.3.0") && nodeVersion.before("6.4.0")) {
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: I'm sure I'm missing something, but wouldn't this only change the version that we try to start on, but not stop the node from attempting to start?

Comment (jkakavas) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: Correct. The intention is to not mute the tests but keep running them even when we're in a FIPS JVM in CI. The way to achieve this is to make sure that older ES version ( not supporting fips ) nodes start with a non fips java version

Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: This is not at all clear from the code. Can you please add a comment explaining it? If I understand correctly, by adding this other elseif condition, non fips testing will fall through and continue using the RUNTIME_JAVA_HOME? But isn't that a fips jvm in this case?

Comment (jkakavas) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: > This is not at all clear from the code. Can you please add a comment explaining it?

Sure thing. I tried to capture it in `// Versions before 6.4.0 cannot be run in a FIPS 140 JVM` but I agree that's not very clear. 

> If I understand correctly, by adding this other elseif condition, non fips testing will fall through and continue using the RUNTIME_JAVA_HOME? But isn't that a fips jvm in this case?

Not sure I follow your thought. When RUNTIME_JAVA_HOME is a fips JVM, `project.inFipsJvm` will also be true. 
In summary: 

- When running in a non FIPS JVM
  - Nothing changes from the previous behavior
  - We run < 6.2.0 nodes with Java 8
  - We run > 6.2.0 and < 6.3.0 nodes with Java 9
  - We run > 6.3.0 nodes with RUNTIME_JAVA_HOME ( non FIPS )

- When running in a FIPS JVM
  - `project.inFipsJvm` is true
  - We run < 6.2.0 nodes with Java 8 (non fips)
  - We run > 6.2.0 and < 6.3.0 nodes with Java 9 (non FIPS)
  - We run > 6.3.0 and < 6.4.0 nodes with Java 10 (non FIPS)
  - We run > 6.4.0 nodes with RUNTIME_JAVA_HOME ( which is FIPS but > 6.4 nodes can run fine in a FIPS JVM)

Does this make more sense ? Bear with me if I've missed your point entirely.

Comment (rjernst) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: Thanks, that is more clear.

Comment (jkakavas) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: @rjernst I updated the comment, let me know if this is clear enough, thanks !

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by jkakavas** [comment]
jenkins test this please

**Comment by jkakavas** [comment]
Jenkins test this please

**Comment by jkakavas** [comment]
I'd like to merge this to get the fips CI green before the weekend. It seems that packaging test runs are still not getting started but the changes in here don't have any impact on packaging. Thoughts @rjernst ?

**Comment by rjernst** [comment]
I think this is ok to merge.

**Pull request merged**